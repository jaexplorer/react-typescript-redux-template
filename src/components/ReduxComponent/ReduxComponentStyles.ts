import styled from 'styled-components/macro';

export interface ReduxComponentProps {
  example?: boolean;
}

export const ReduxComponentContainer = styled.div<ReduxComponentProps>``;

export const Loading = styled.div``;

export const Content = styled.div``;

export const ExampleItem = styled.div``;
