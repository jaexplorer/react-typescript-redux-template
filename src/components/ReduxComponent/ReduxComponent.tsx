import React, { FC, useEffect } from 'react';
import { connect } from 'react-redux';
import { Dispatch, bindActionCreators } from 'redux';
import { ApplicationState } from '../../lib/Redux/Store';
import { Example } from '../../lib/Types';
import { loadExample, LoadExampleResult } from '../../lib/Redux/Actions/example';
import { ReduxComponentContainer, Loading, Content, ExampleItem } from './ReduxComponentStyles';

const ReduxComponent: FC<ReduxComponentProps> = ({ loading, examples, onLoadExample }) => {
  useEffect(() => {
    onLoadExample('123');
  }, []);

  return (
    <ReduxComponentContainer>
      {loading ? (
        <Loading />
      ) : (
        <Content>
          {examples.map((item) => (
            <ExampleItem key={item.id} />
          ))}
        </Content>
      )}
    </ReduxComponentContainer>
  );
};

export interface ReduxComponentProps {
  loading: boolean;
  examples: Example[];
  onLoadExample: (exampleId: string | number) => LoadExampleResult;
}

const mapStateToProps = ({ loadingState, exampleState }: ApplicationState) => ({
  loading: loadingState.apiLoading,
  examples: exampleState.examples,
});

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      onLoadExample: loadExample as any,
    },
    dispatch,
  );

export default connect(mapStateToProps, mapDispatchToProps)(ReduxComponent);
