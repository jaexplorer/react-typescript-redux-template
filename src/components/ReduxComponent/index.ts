import ReduxComponent from './ReduxComponent';

export default ReduxComponent;
