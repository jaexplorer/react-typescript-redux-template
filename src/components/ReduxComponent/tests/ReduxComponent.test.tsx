import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from '../../../lib/Redux/Store';
import ReduxComponent from '../ReduxComponent';

test('renders component', () => {
  const store = configureStore();

  const { container } = render(
    <Provider store={store}>
      <ReduxComponent />
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
