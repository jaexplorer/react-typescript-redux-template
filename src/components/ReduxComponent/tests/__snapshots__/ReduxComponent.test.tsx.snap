// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
<div>
  <div
    class="ReduxComponentStyles__ReduxComponentContainer-sc-1ufayrv-0 cREbXH"
  >
    <div
      class="ReduxComponentStyles__Loading-sc-1ufayrv-1 jDtfMS"
    />
  </div>
</div>
`;
